airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeCopyGcsDestinationTest.java

```diff
@@ -0,0 +1,50 @@
+package io.airbyte.integrations.destination.snowflake;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.resources.MoreResources;
+import io.airbyte.integrations.destination.ExtendedNameTransformer;
+import io.airbyte.integrations.destination.jdbc.SqlOperations;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class SnowflakeCopyGcsDestinationTest {
+
+  final SnowflakeCopyGcsDestination copyGcsDestination = new SnowflakeCopyGcsDestination();
+  JsonNode config;
+
+  @BeforeEach
+  public void setup() throws IOException {
+    config = spy(Jsons.deserialize(MoreResources.readResource("insert_config.json"), JsonNode.class));
+  }
+
+  @Test
+  void getNameTransformer() {
+    ExtendedNameTransformer actualNameTransformer = copyGcsDestination.getNameTransformer();
+    assertEquals(actualNameTransformer.getClass(), SnowflakeSQLNameTransformer.class);
+  }
+
+  @Test
+  void getDatabase() throws Exception {
+    copyGcsDestination.getDatabase(config);
```

Comment:
Regarding this case - unfortunately, we cannot check output of getDatabase() method as far as DataSource instance has protected access level in DefaultJdbcDatabase that extends JdbcDatabase, we can change access level, but it is not common approach change code for test perspective.
Regarding other cases and testS3ConfigType case we cannot check output, because state is encapsulated and open api is out of unit test. 
I agree with you that it is not elegant unit tests, but in code we use static methods and create new object inside methods and this makes it difficult to write the correct unit tests.

Reply:
ok. I think I'd rather just delete these tests entirely then, they're probably not really adding much value :/

Actually - there are some cases where we expose a method for the sake of tests (tagging methods with `@VisibleForTesting`). Would that be helpful here?